src/protocol/handshake.rs

```diff
@@ -117,50 +167,49 @@ impl Handshake for InnerNode {
+                    error!(parent: self.node().span(), "missing Sec-WebSocket-Key");
                     return Err(io::ErrorKind::InvalidData.into());
                 };
 
                 let mut rsp = Vec::new();
-                rsp.extend_from_slice(b"HTTP/1.1 101 Switching Protocols\r\n");
-                rsp.extend_from_slice(b"Upgrade: websocket\r\n");
-                rsp.extend_from_slice(b"Connection: Upgrade\r\n");
-                rsp.extend_from_slice(format!("Sec-Websocket-Accept: {}\r\n", swa).as_bytes());
-                rsp.extend_from_slice(
-                    format!("X-Algorand-Instancename: {}\r\n", X_AG_INSTANCE_NAME).as_bytes(),
-                );
-                rsp.extend_from_slice(b"X-Algorand-Location:\r\n");
-                rsp.extend_from_slice(
-                    format!("X-Algorand-Noderandom: {}\r\n", X_AG_NODE_RANDOM).as_bytes(),
-                );
-                rsp.extend_from_slice(
-                    format!("X-Algorand-Version: {}\r\n", X_AG_ACCEPT_VERSION).as_bytes(),
-                );
-                rsp.extend_from_slice(
-                    format!("X-Algorand-Genesis: {}\r\n", X_AG_ALGORAND_GENESIS).as_bytes(),
-                );
-                if let Some(challenge) = &self.challenge {
-                    rsp.extend_from_slice(
-                        format!("X-Algorand-Prioritychallenge: {challenge}\r\n").as_bytes(),
-                    );
+                let mut rsp_header = |mut header: String| {
+                    header.push_str("\r\n");
+                    rsp.extend_from_slice(header.as_bytes());
+                };
+
+                rsp_header("HTTP/1.1 101 Switching Protocols".into());
+                rsp_header("Upgrade: websocket".into());
+                rsp_header("Connection: Upgrade".into());
+                rsp_header(format!("Sec-Websocket-Accept: {swa}"));
+                rsp_header(format!("X-Algorand-Instancename: {}", cfg.ar_instance_name));
+                if let Some(ref location) = cfg.ar_location {
+                    rsp_header(format!("X-Algorand-Location: {location}"));
                 }
-                rsp.extend_from_slice(b"\r\n");
-                let rsp = Bytes::from(rsp);
+                rsp_header(format!("X-Algorand-Noderandom: {}", cfg.ar_node_random));
+                rsp_header(format!("X-Algorand-Version: {}", cfg.ar_accept_version));
+                rsp_header(format!("X-Algorand-Genesis: {}", cfg.ar_genesis));
+                if let Some(ref challenge) = cfg.challenge {
+                    rsp_header(format!("X-Algorand-Prioritychallenge: {challenge}"));
+                }
+                rsp_header("".into()); // A HTTP header ends with '\r\n'
 
-                info!(parent: self.node().span(), "sending handshake message: {:?}", rsp);
+                let rsp = Bytes::from(rsp);
+                info!(parent: self.node().span(), "sending handshake a handshake response: {:?}", rsp);
```

Comment:
Shouln't be "sending a handshake response" ?

Reply:
Done